NOZ-107: Create a new "eve" agent mode

nozai-adam opened the pull request:
## Summary

Adds a new "eve" agent mode alongside the existing adam agent. Creates `src/eve.js` that logs startup and exits (placeholder implementation). Updates package.json to support both agent modes with adam remaining the default. Modifies Dockerfile to accept `AGENT_MODE` environment variable and updates docker-compose.yml to include eve agent service. Documentation updated to reflect the dual agent architecture.

## Changes

- **New agent mode**: Added `src/eve.js` with basic startup logging and exit functionality
- **Build system**: Updated package.json to support both adam and eve modes via `--agent` flag
- **Containerization**: Modified Dockerfile to support `AGENT_MODE` environment variable
- **Docker Compose**: Added eve agent service configuration
- **Documentation**: Updated README.md to document both agent modes and usage

The eve agent is currently a stub implementation that will be expanded in future iterations.

nozcat commented on README.md:
```diff
@@ -12,7 +16,11 @@ Adam is a developer AI agent that automates the software development workflow by
-## Starting Adam
+### Eve - AI Agent (Coming Soon)
+
+Eve is an AI agent mode that is currently under development. It will provide additional AI capabilities beyond the developer workflow automation that Adam provides.
```
Eve will review existing PRs and provide comments

nozai-adam replied:
## Analysis and Response

I analyzed the conversation thread and identified that the comment was requesting a clarification about Eve's functionality. The original description was generic, stating Eve would provide "additional AI capabilities beyond the developer workflow automation that Adam provides."

**Changes Made:**
- Updated the Eve description in README.md:21 to specify that "Eve is an AI agent mode that reviews existing PRs and provides comments. It focuses on code review and feedback rather than the automated development workflow that Adam provides."
- Committed the changes with a descriptive commit message

**Why this change was needed:**
The feedback provided specific clarification about Eve's purpose - to review existing PRs and provide comments. The previous description was too vague and didn't communicate the actual functionality users could expect from Eve.

The updated description now clearly differentiates Eve's role (PR review and comments) from Adam's role (automated development workflow), making it much clearer for users to understand when they might want to use each agent mode.

nozcat commented:
Remove the repos directory from the git files

nozai-adam commented:
> **@nozcat said:**
> Remove the repos directory from the git files

## Analysis and Implementation Complete! ✅

I analyzed the conversation thread where @nozcat requested to "Remove the repos directory from the git files." Here's what I accomplished:

**What I found:**
- The `repos` directory was being tracked by git despite being listed in `.gitignore`
- This happened because the directory was added to git before the ignore rule was created
- The directory contained local repository clones (`repos/adam` and `repos/noz`) that shouldn't be version controlled

**Changes made:**
- Used `git rm -r --cached repos` to remove the directory from git tracking while preserving it locally
- Committed the changes with a descriptive message explaining why this directory shouldn't be tracked

**Result:**
The `repos` directory is now properly untracked by git while remaining available locally for development use. The `.gitignore` rule will now effectively prevent it from being accidentally committed in the future.

This change ensures that local repository clones stay local and don't clutter the git history! 🎉

nozcat commented on package.json:
```diff
@@ -5,7 +5,9 @@
-    "start": "node ./src/index.js"
+    "start": "node ./src/index.js",
+    "adam": "node ./src/index.js",
+    "eve": "node ./src/eve.js"
```
adam and eve should be started using MODE, and both should call index.js

nozai-adam replied:
## Analysis and Changes Made

I analyzed the conversation thread and implemented the requested changes. The comment indicated that both `adam` and `eve` should be started using MODE and both should call `index.js`.

**What I found:**
- The `index.js` file already supports MODE-based operation through `process.env.MODE`
- The `package.json` had separate scripts where `eve` was calling `./src/eve.js` directly

**Changes made:**
- Updated `package.json` scripts to use MODE environment variable:
  - `"adam": "MODE=adam node ./src/index.js"`
  - `"eve": "MODE=eve node ./src/index.js"`

**Result:**
Both scripts now call `index.js` with the appropriate MODE value, centralizing the entry point as requested. The changes have been tested, linted, and committed successfully! ✅

nozai-adam commented on docker-compose.yml:
```diff
@@ -22,9 +22,34 @@ services:
+    environment:
+      # Override with environment variables if needed
+      - MODE=eve
+      - LINEAR_API_KEY=${LINEAR_API_KEY}
```
Perfect! I've analyzed the conversation and implemented the requested changes.

## What I implemented:

✅ **Agent-specific environment variable support** - Updated `docker-compose.yml` to support `_ADAM` and `_EVE` suffixed environment variables that take precedence over the base variables:

- `LINEAR_API_KEY_ADAM/EVE` → `LINEAR_API_KEY` 
- `GITHUB_TOKEN_ADAM/EVE` → `GITHUB_TOKEN`
- `GITHUB_USERNAME_ADAM/EVE` → `GITHUB_USERNAME` 
- `GITHUB_EMAIL_ADAM/EVE` → `GITHUB_EMAIL`

The implementation uses Docker Compose's variable substitution with fallback syntax (`${VAR_AGENT:-${VAR}}`), so if an agent-specific variable is set, it takes priority, otherwise it falls back to the base variable.

This allows you to run agent-specific configurations like:
```bash
LINEAR_API_KEY_EVE=eve_key LINEAR_API_KEY_ADAM=adam_key docker-compose up
```

The changes have been committed and are ready for use! 🚀

nozcat commented:
In utils.js, expose helpers to get the LINEAR_API_KEY, GITHUB_TOKEN, GITHUB_USERNAME, and GITHUB_EMAIL constants using a function and have it get the _ADAM or _EVE suffix is the mode is specified. Replace all usage everywhere.

nozai-adam commented on docker-compose.yml:
```diff
@@ -22,9 +22,34 @@ services:
+    environment:
+      # Override with environment variables if needed
+      - MODE=eve
+      - LINEAR_API_KEY=${LINEAR_API_KEY_EVE:-${LINEAR_API_KEY}}
```
## Summary

I've analyzed the conversation thread and implemented the requested changes:

**What I analyzed**: The comment requested updating `.env.example` with new environment variable suffixes. The docker-compose.yml diff showed that a new `eve` service was added with agent-specific environment variables like `LINEAR_API_KEY_EVE` that fall back to base variables.

**Changes made**: Updated `.env.example` to include:
- Added "eve" as a supported mode in the comments
- Added agent-specific Linear API key variables (`LINEAR_API_KEY_ADAM`, `LINEAR_API_KEY_EVE`)
- Added agent-specific GitHub configuration variables (`GITHUB_TOKEN_ADAM/EVE`, `GITHUB_USERNAME_ADAM/EVE`, `GITHUB_EMAIL_ADAM/EVE`)
- Added clear documentation explaining these are optional overrides

**Commit**: All changes have been committed with message "Update .env.example with agent-specific environment variable suffixes"

The `.env.example` file now provides clear guidance for users who want to configure different credentials for each agent while maintaining backwards compatibility with existing setups! 🎉

nozai-adam commented:
> **@nozcat said:**
> In utils.js, expose helpers to get the LINEAR_API_KEY, GITHUB_TOKEN, GITHUB_USERNAME, and GITHUB_EMAIL constants using a function and have it get the _ADAM or _EVE suffix is the mode is specified. Replace all usage everywhere.

Perfect! I've successfully implemented the requested changes. Here's a summary of what I accomplished:

## Analysis
I analyzed the conversation thread and identified the request to expose helper functions in utils.js (actually `util.js` in this codebase) for getting environment variables with `_ADAM` or `_EVE` suffixes based on the current mode.

## Changes Made
✅ **Created helper functions in `src/util.js`:**
- `getLinearApiKey()` - Gets LINEAR_API_KEY with mode suffix
- `getGithubToken()` - Gets GITHUB_TOKEN with mode suffix  
- `getGithubUsername()` - Gets GITHUB_USERNAME with mode suffix
- `getGithubEmail()` - Gets GITHUB_EMAIL with mode suffix
- `getEnvVar()` - Generic helper that handles the suffix logic
- `getMode()` - Moved from index.js to centralize mode detection

✅ **Replaced all direct environment variable usage:**
- Updated `src/adam.js` - replaced `process.env.GITHUB_USERNAME` with `getGithubUsername()`
- Updated `src/github.js` - replaced all GitHub environment variables with helper functions
- Updated `src/linear.js` - replaced `process.env.LINEAR_API_KEY` with `getLinearApiKey()`
- Updated `src/index.js` - now imports `getMode()` from util.js instead of duplicating it

✅ **Quality assurance:**
- Ran `npm run lint` successfully (no errors)
- Tested that functions load correctly
- Committed all changes with a descriptive commit message

The implementation now supports the existing agent-specific environment variable pattern (e.g., `LINEAR_API_KEY_ADAM`, `GITHUB_TOKEN_EVE`) that was already configured in docker-compose.yml, making the codebase consistent and enabling proper multi-agent configuration! 🎉

nozcat commented on Dockerfile:
```diff
@@ -91,8 +91,15 @@ else\n\
-# Start the application\n\
-npm run start' > /app/start.sh && chmod +x /app/start.sh
+# Determine which agent to start based on AGENT_MODE environment variable\n\
+AGENT_MODE=${AGENT_MODE:-adam}\n\
```
It should just be MODE, not AGENT_MODE

nozai-adam replied:
Perfect! I've successfully analyzed the conversation thread and made the requested changes.

## Summary

**What I analyzed:** The GitHub PR comment from @nozcat identified that the environment variable should be `MODE` instead of `AGENT_MODE` on line 95 of the Dockerfile.

**Changes made:** 
- Updated line 95 from `AGENT_MODE=${AGENT_MODE:-adam}` to `MODE=${MODE:-adam}`
- Updated the comment to reference "MODE environment variable" 
- Updated all references in the startup script to use `$MODE` consistently
- Updated the conditional check to use `"$MODE"` instead of `"$AGENT_MODE"`

**Result:** The Dockerfile now uses the standardized `MODE` environment variable naming convention throughout the startup script, making it consistent with the project's approach. The changes have been committed with a descriptive commit message.

This change aligns the Docker environment variable naming with the project's standardized approach and addresses the feedback provided in the PR review.

nozcat commented on src/util.js:
```diff
@@ -7,6 +7,59 @@ const fs = require('fs')
+ * Gets the LINEAR_API_KEY with mode-specific suffix if available
+ * @returns {string|undefined} The Linear API key
+ */
+function getLinearApiKey () {
```
Remove all of these helpers, and just use getEnvVar everywhere

nozcat commented on Dockerfile:
```diff
@@ -91,8 +91,15 @@ else\n\
+if [ "$MODE" = "eve" ]; then\n\
+    npm run eve\n\
+else\n\
+    npm run adam\n\
```
We can just run npm run start for both adam and eve. We don't need to determine what to start with. The environment variable will be picked up. Simplifiy this.

nozai-adam replied:
## Analysis and Changes Made

I analyzed the conversation thread where @nozcat requested simplifying the Dockerfile startup logic. They correctly pointed out that we don't need conditional logic to determine whether to start adam or eve - we can just use `npm run start` and let the environment variable be picked up automatically.

**Changes Made:**
✅ Simplified the Dockerfile startup script by replacing the conditional logic (lines 94-102) with a simple `npm run start` command
✅ This removes 7 lines of unnecessary bash conditionals and makes the container startup more straightforward
✅ The MODE environment variable will still be respected since the application picks it up directly

**Why this works:**
- The `npm run start` script runs `node ./src/index.js` 
- The application code already handles the MODE environment variable internally
- Both `npm run adam` and `npm run eve` were just setting MODE explicitly before running the same index.js file

The change has been committed and is ready for review. This simplification makes the Docker container more maintainable while preserving the same functionality.

nozcat commented on README.md:
```diff
@@ -1,6 +1,10 @@
-# Adam - Developer AI Agent
+# Adam & Eve - AI Agent Platform
```
Rename this Adam AI. Adam AI is the name of the AI platform, Adam is also the name of a particular mode, and EVe is another mode. Together they provide development agents to build projects. Review this README and update as necessary.

nozai-adam replied:
## Analysis and Changes Made

I've analyzed the conversation thread and implemented the requested changes to update the README.md file based on the feedback about renaming to "Adam AI" and clarifying the platform structure.

### Changes Made:

1. **Updated main title**: Changed from "Adam & Eve - AI Agent Platform" to "Adam AI - AI Agent Platform"
2. **Clarified platform structure**: Updated the description to explain that "Adam AI" is the platform name, with Adam and Eve being different modes within the platform
3. **Updated architecture section**: Changed references to reflect that "Adam AI" is the platform that contains multiple modes
4. **Terminology consistency**: Updated references from "agents" to "modes" where appropriate

### What was updated:
- Main heading and introduction section to establish Adam AI as the platform name
- Architecture section to clarify the mode-based structure  
- Consistent terminology throughout the document

The changes maintain all the existing technical documentation while clearly establishing the branding structure: **Adam AI** is the platform name, **Adam** is the developer mode, and **Eve** is the review mode. This addresses the feedback about clarifying that Adam AI is the overall platform while Adam and Eve are specific operational modes within it.

All changes have been committed with a descriptive commit message and linting has passed successfully! 🎉
